zeebe/scheduler/src/main/java/io/camunda/zeebe/scheduler/Actor.java

```diff
@@ -99,6 +101,9 @@ public void close() {
 
   @Override
   public ActorFuture<Void> closeAsync() {
+    if (metrics != null) {
+      metrics.close();
+    }
```

Comment:
:x: We should close the metrics as the last thing the actor does. This is too early - this is when requesting the close, and can be called from a different thread. Instead, the actor task should take care of closing the metrics. Basically, we should close it when we're transition the actor lifecycle state to closed.

Reply:
I moved it to `ActorTask.onClose()`, is that the right place?

Reply:
I think so yes :+1: